### ISSUE TITLE
Query Log empty and Graphs not updating (not on Jesse)

### ISSUE THREAD

**Issue by 1activegeek** [pi-hole/pi-hole#2270]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`2`

---
**Expected behaviour:**

`Ads being blocked on devices, Query log to show recent queries, and Graphs to show current data`

**Actual behaviour:**

`Ads being blocked, Query log is empty after running for a little while, and graphs also notch down to stopping after awhile.`

**Steps to reproduce:**

`Run piHole for a few hours - check console`

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`Use netcat.`

Seemed odd, but that was all that was output:
<img width="535" alt="screen shot 2018-07-09 at 11 21 41 pm" src="https://user-images.githubusercontent.com/6810720/42487423-f69d840a-83ce-11e8-935e-008eb709dd53.png">

**Troubleshooting undertaken, and/or other relevant information:**

`Simply restarting the service or Dnsmasq will not resolve the issue. Restarting the system will start everything again. Flushing the log and restarting Dnsmasq/pihole services will start everything again.`
`System is Rock64 aarch64 proc and running DietPi 6.11. Also of note, it seems the block page is not working correctly either - Block page is a single white page with one line stating "Blocked by PiHole", no option to override with password or anything like it used to.`

```
root:~# cat /etc/*release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

root:~# dnsmasq -v
Dnsmasq version 2.76  Copyright (c) 2000-2016 Simon Kelley
Compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset auth DNSSEC loop-detect inotify

root:~# pihole -v
  Pi-hole version is v3.3.1 (Latest: v3.3.1)
  AdminLTE version is v3.3 (Latest: v3.3)
  FTL version is v3.0 (Latest: v3.0)
```

**Comment by dschaper** [comment]
First lets take a look at the `netcat` error so we can get a look at the debug log. Do you have the `openssl` package installed in DietPi? What is the output from `which openssl`?

**Comment by 1activegeek** [comment]
Sorry for the delay. Sure thing. I do have openssl installed I believe, output:
`/usr/bin/openssl`

**Comment by dschaper** [comment]
No problem on the delay, did you install via the DietPi programs or via `curl https://install.pi-hole.net`?

**Comment by 1activegeek** [comment]
Via DietPi install as I recall

**Comment by dschaper** [comment]
The reasons I ask is that if `openssl` is installed you should not fall back to `nc` for the upload, we check https://github.com/pi-hole/pi-hole/blob/master/advanced/Scripts/piholeDebug.sh#L1075 to see. Can you try `pihole -d` again and see if it gives you the same error, and if it does, can you check `/opt/pihole/piholeDebug.sh` and see if you have the same contents of that file as I have linked?

**Comment by 1activegeek** [comment]
Interesting - this time I got a new error which may be part of issue before, but still no bueno.
```
*** [ DIAGNOSING ]: Pi-hole log
-rw-r--r-- 1 dnsmasq www-data 2 Jul 12 00:00 /var/log/pihole.log
   -----head of pihole.log------
    

cat: write error: No space left on device

********************************************
********************************************
[✓] ** FINISHED DEBUGGING! **

    * The debug log can be uploaded to tricorder.pi-hole.net for sharing with developers only.
    * For more information, see: https://pi-hole.net/2016/11/07/crack-our-medical-tricorder-win-a-raspberry-pi-3/
    * If available, we'll use openssl to upload the log, otherwise it will fall back to netcat.

[?] Would you like to upload the log? [y/N] n
    * Log will NOT be uploaded to tricorder.
root@odyssey:~# df
Filesystem     1K-blocks    Used Available Use% Mounted on
udev             2007908       0   2007908   0% /dev
tmpfs             401960   40928    361032  11% /run
/dev/mmcblk1p7  15118828 9008132   5629744  62% /
tmpfs            2009784       0   2009784   0% /dev/shm
tmpfs               5120       0      5120   0% /run/lock
tmpfs            2009784       0   2009784   0% /sys/fs/cgroup
tmpfs              20480   20480         0 100% /var/log
tmpfs              10240    1476      8764  15% /DietPi
tmpfs            2009088       0   2009088   0% /tmp
/dev/mmcblk1p6    102182   42614     59568  42% /mnt/6393-85D3
```

You'll notice my /var/log is out of space ... 🙄 might have to ping fourdee and see how to adjust/fix that. I think something could be wrong here which could be the cause of the issue overall if no space left. Syslog is at 7.4M, looks like it's full of dnsmasq query lines. Think I just found a can of worms!

**Comment by dschaper** [comment]
Okay, we talk with @Fourdee too and I'll follow this to see what comes up. I'm looking at testing DietPi myself with a few new boards to see how it goes so I'm interested to see what comes of it. We're rolling out some changes coming up with regards to logging and trying to cut down on our logging verbosity and the number of writes to the SDCards.

**Comment by Fourdee** [comment]
@1activegeek @dschaper 

Apologies for the delay, only just come back to DietPi and catching up on whats been our end.

Ok, so it looks like there was an issue with v6.11, that caused DietPi cron jobs to fail, (fixed by @MichaIng 
 👍 ). 
I believe this would also render the clearing of logs failed (hence the full size log directory), with RAMlog mode 1.

V6.12 should be out next week, but in the mean time, please try the following fix:
https://github.com/Fourdee/DietPi/issues/1923#issue-341325599
```
https://raw.githubusercontent.com/Fourdee/DietPi/0f18aa4dc0af8ab910a0173dce8849d5b53c30b0/dietpi/func/dietpi-globals -O /DietPi/dietpi/func/dietpi-globals
```

And to clear the current log files:
```
/DietPi/dietpi/dietpi-logclear 1
```

---

#### More info on our Pi-Hole logging:
- Max logfile size = 5M. If the file size is above this, the log will be cleared.
- Non-today entries are automatically cleared.

Aside from that, in general usage, you should have enough space (5M cap) for general usage of Pi-Hole, and, stored daily entries in the logs for web query.

From our end, i'll see if we can increase the tmpfs size to 50MB and PiHole log max to 20MB for high usage.

**Comment by Fourdee** [comment]
@dschaper 

This ticket should all be resolved in our next update v6.12.
- 16MB pihole logfile limit (up from 5MB)
- Resolved DietPi-RAMlog not clearing down the logs via cron correctly.

ETA next week release. Until then, workaround to resolve full logs = https://github.com/pi-hole/pi-hole/issues/2270#issuecomment-405109135

**Comment by dschaper** [comment]
Thanks so much @Fourdee ! In some updates that we are working on we'll address the logfile issue. I know that there are a lot of users that don't like the writes to SD or large logfiles so we're trying to fix that issue.

**Comment by 1activegeek** [comment]
@Fourdee @dschaper thanks guys ... you all rock!! I'll see if I can't run through a test install again. To get past the WAF I had to move my piHole back to the docker container. I'll try setting it back up and run the tests.

**Comment by 1activegeek** [comment]
Update: So far so good. Looks like it's holding well this time around. I can see that `/var/log` is 20M and currently after about 2 hrs of running, the usage is only at 3.8M. So it looks like rotation and such is happening as expected, and that the admin console still shows updating graphs and the query log is still populating queries. Will keep an eye on it over the coming days to validate no further issue. 

Next up - testing out the cloudflared option again for DoH! 😄

**Comment by 1activegeek** [comment]
Ok ... new update ... seems there is something going wrong now with slow DNS resolution. At first it seemed fine, but now I load things like gitlab.com, amazon.com, even google.com and it starts to load in browser, then shows the halfway ... then just hangs. I have to reload/refresh or retry and sometimes even that isn't working either. I noticed this before when I first moved onto Rock64 standalone, but I thought it was related to the DoH config, but I've not set that up yet. Anything I can do to help dig into this issue now?

**Comment by dschaper** [comment]
F12 the page in the browser and see what asset/element is not being loaded or painted. I don't think this is DNS related as by the time the elements are being displayed the DNS is done and it's just the assets being streamed in.

**Comment by 1activegeek** [comment]
Unfortunately when it was happening yesterday I was on the way out the door so couldn't get any more details. It seems free from this for now ... if it creeps up again, I'll open a new issue and just reference this one for any historical needs. 😄
